lib/system.nim

```diff
@@ -4210,6 +4210,23 @@ when hasAlloc or defined(nimscript):
       x[j+i] = item[j]
       inc(j)
 
+  func insert*(x: var string, c: char, i = 0.Natural) =
+    ## Inserts character `c` into `x` at position `i`
+    ##
+    ## .. code-block:: Nim
+    ##   var a = "hllo"
+    ##   a.insert('e', 1) # a <- "Hello"
+    var xl = x.len
+    setLen(x, xl+1)
+    var j = xl-1
+    while j >= i:
+      shallowCopy(x[j+1], x[j])
```

Comment:
Shouldn't this be `x[j+1] = x[j]`? No point in using shallowCopy on char.

Reply:
I agree, 2 `while` with `shallowCopy` on it I dunno if its the best algo tho.

Reply:
No, shallowCopy made no sense, this came from the (string, string) implementation. I changed into a simple assignment now. I guess use copyMem() or moveMem() would improve performance, but afaik this is not available on the JS backend?